### ISSUE TITLE
DELTA_GRPC used for GRPC api type

### ISSUE THREAD

**Issue by kyessenov** [envoyproxy/envoy#8927]
I'm seeing delta xDS requests sent for CDS set to use regular GRPC api_type:
```
[2019-11-06 19:17:34.029][19268][info][upstream] [source/common/upstream/cluster_manager_impl.cc:156] cm init: initializing cds
[2019-11-06 19:17:34.030][19268][debug][config] [bazel-out/k8-fastbuild/bin/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:47] Establishing new gRPC bidi stream for rpc DeltaClusters(stream .envoy.api.v2.DeltaDiscoveryRequest) returns (stream .envoy.api.v2.DeltaDiscoveryResponse);
```

This is for the bootstrap https://github.com/envoyproxy/go-control-plane/blob/master/sample/bootstrap-xds.yaml

**Comment by kyessenov** [comment]
The effect is that the config push is stalled since the server does not implement DELTA_GRPC method.

**Comment by mattklein123** [comment]
Another likely regression from https://github.com/envoyproxy/envoy/commit/443bc3345b0e3db99a3df03d52f317697b99d5d7. cc @fredlas @wgallagher. 

We will likely revert that commit tomorrow if so as there is another issue reported.

**Comment by fredlas** [comment]
What version of Envoy? This sounds at least vaguely similar to the problem that #8587 solved (I don't think the problem itself ever got an issue filed I can link to).

Looking at the new code, a GrpcStream<Delta> is only ever held by a GrpcMuxDelta. The only places those are created are https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/cluster_manager_impl.cc#L246 and https://github.com/envoyproxy/envoy/blob/master/source/common/config/subscription_factory_impl.cc#L61  , both of which are directly under a conditional explicitly checking for the value ApiConfigSource::DELTA_GRPC (as opposed to being an `else` or `default`). So, I would be extremely surprised if a config that does not contain a "DELTA_GRPC" managed to hit either of those. It would have to be some error in the reading+parsing of the bootstrap config, which I don't think my changes ever touched.

**Comment by kyessenov** [comment]
@fredlas This is a build from master (commit c2eae094ce28b90861f785762f0f9d5ecfcd837c). Just try to run the provided bootstrap and observe the delta request.

**Comment by kyessenov** [comment]
@fredlas I built envoy right before https://github.com/envoyproxy/envoy/commit/443bc3345b0e3db99a3df03d52f317697b99d5d7 and it doesn't invoke Delta calls.

**Comment by mattklein123** [comment]
I'm inclined to revert the big PR given this and the other issue, until we sort this out. Any objections @fredlas @wgallagher?

**Comment by fredlas** [comment]
Did you try rebuilding right at https://github.com/envoyproxy/envoy/commit/443bc3345b0e3db99a3df03d52f317697b99d5d7 ? Just in case the earlier one you were running was not built at the version you thought? I know that's unlikely, but as I described, there are some fairly straightforward parts of the code that make this all seem impossible.

**Comment by fredlas** [comment]
(I will also give running it a try)

**Comment by kyessenov** [comment]
Yeah, https://github.com/envoyproxy/envoy/commit/443bc3345b0e3db99a3df03d52f317697b99d5d7 is the commit when it starts calling Delta RPCs.

**Comment by kyessenov** [comment]
@fredlas You might have inadvertently activated some latent bug. It's hard to tell but easy to reproduce, so should be fairly easy to trace the root cause.

**Comment by fredlas** [comment]
Ahhhhh ok, tried it myself and saw the error. I appreciate the charitable interpretation of possibly uncovering a latent bug! That was certainly my next guess. But, I read the code over again, and found a really stupid copy+paste mistake in my own code: there is a utility function that retrieves the gRPC method descriptor, with one version for delta and one for SotW, and I copied the delta twice. Whoops, sorry :(

I have verified that fixing it fixes the immediate bad behavior. I think that should do it. I will leave it up to you all whether you want to give fixing forward with this a try, or just revert first. I'll open a PR for it shortly.
